crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -1420,6 +1420,30 @@ unsafe impl<T: SystemBuffer> SystemParam for Deferred<'_, T> {
     }
 }
 
+/// A dummy type that is [`!Send`](Send), to force systems to run on the main thread.
+pub struct NonSendMarker();
```

Comment:
This could be a little simpler as a unit struct instead of a tuple struct:
```suggestion
pub struct NonSendMarker;
```